### ISSUE TITLE
Base rule extension: [max-params] don't count this: void

### ISSUE THREAD

**Issue by RettentoRectangle** [typescript-eslint/typescript-eslint#7538]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+base+rule+extension%22) and found none that match my proposal.
- [X] I have searched the [current extension rule list](https://typescript-eslint.io/rules/#extension-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the base rule

https://eslint.org/docs/latest/rules/max-params

### Description

It would be great ignore "this: void" by default.

Eslint cries about "this: void" even though it is not a real parameter.

### Fail

```typescript
// Fail. Still report an error if too much parameters are there:

private doSomethingWild (x: number, y: number, z: number, k: any) {
 //
}
```

### Pass

```typescript
// Pass. Don't count "this: void".

private doSomethingWild (this: void, x: number, y: number, z: number) {
 //
}
```

### Additional Info

I don't have time to make it myself. :(

**Comment by RettentoRectangle** [comment]
Ups, forget to name it after filled the template.
